src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -110,7 +110,8 @@ public static void setup(Level level, String logLocation, boolean logFormatRfc33
 
     /** Laconic logging for reduced verbosity. */
     public static void laconic(org.slf4j.Logger logger, Level level, String message, int max) {
-        if (messageCount.count(message) <= max) {
+        int messageCount = getAndIncrementMessageCount(message);
+        if (messageCount <= max) {
```

Comment:
previously we only incremented the count if it was still below the maximum (ie. check then increment) whereas we now always increment the count - in theory this risks overflowing the integer, but in practice it's likely not an issue